src/models/minecraft_protocol.rs

```diff
@@ -281,8 +281,8 @@ fn read_chunk_section<S: Read>(stream: &mut S) -> ChunkSection {
         if bits_to_read < 14 {
             let remainder_to_read = 14 - bits_to_read;
             let remainder = long << (64 - remainder_to_read) >> (64 - remainder_to_read);
-            block_id += remainder >> remainder_to_read;
-        };
+            block_id += remainder << bits_to_read;
```

Comment:
Weird. Well if it works then it works

Reply:
I think it's because the remainder is in little endian and the remainder_to_read variable is only needed to shift left and right, we are still reading 'bits_to_read' bits.